### PR TITLE
Update dependencies, use Stac4s

### PR DESCRIPTION
**Pull request by notthatbreezy**
This PR updates dependencies to their latest version and also switches to using `stac4s` instead of `geotrellis-server`.

### Testing

I think tests should be enough for this, I think I was able to avoid any substantive changes

### CODE REVIEW

File: application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala [hunk 1]
```diff
@@ -17,6 +17,8 @@ object SearchEndpoints {
       .description("Root Catalog and entrypoint into STAC collections")
       .name("root")
 
+  implicit val searchFiltersValidator: Validator[SearchFilters] = Validator.pass[SearchFilters]
```
Comment (notthatbreezy) [inline application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala]: I don't really know why I needed to do this - the implicit codec for `SearchFilters` is found without any trouble, but this won't compile with this.

For now we have a validator that always succeeds ...

File: application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala [hunk 2]
```diff
@@ -2,11 +2,12 @@ package com.azavea.franklin.crawler
 
 import java.nio.file.Paths
 
+import cats.effect.Blocker
 import cats.{Applicative, MonadError}
 import com.azavea.franklin.database.{StacCollectionDao, StacItemDao}
 import doobie.ConnectionIO
 import doobie.implicits._
-import geotrellis.server.stac._
+import com.azavea.stac4s._
```
Comment (jisantuc) [inline application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala]: I wonder if we should use this `scalafix` rule: https://github.com/NeQuissimus/sort-imports

Sorting imports manually is a drag, and it looks like with the `blocks` configuration we would be able to do first party, third party, scala lang, and java lang blocks

File: application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala (hunk 1 above)
Comment (jisantuc) [reply application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala]: It's sort of insane that this works:

```suggestion
  implicit val searchFiltersValidator: Validator[SearchFilters] = implicitly[Validator[SearchFilters]]
```

It makes me wonder if we have some horrible incompatibility that hasn't bitten us in worse ways than this yet :thinking:

File: application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala (hunk 2 above)
Comment (notthatbreezy) [reply application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala]: oh I like this, I'll add it - good find!

File: application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala (hunk 1 above)
Comment (notthatbreezy) [reply application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala]: When I do this I get a runtime error...

```
[E] java.lang.ExceptionInInitializerError
[E]     at com.azavea.franklin.api.Server$.$anonfun$createServer$3(Server.scala:61)
[E]     at cats.effect.Resource.$anonfun$use$1(Resource.scala:121)
[E]     at cats.effect.internals.IOBracket$BracketStart.liftedTree1$1(IOBracket.scala:79)
[E]     at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:79)
[E]     at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:70)
[E]     at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:36)
[E]     at cats.effect.internals.TrampolineEC$JVMTrampoline.super$startLoop(TrampolineEC.scala:93)
[E]     at cats.effect.internals.TrampolineEC$JVMTrampoline.$anonfun$startLoop$1(TrampolineEC.scala:93)
[E]     at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[E]     at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
[E]     at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:93)
[E]     at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
[E]     at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:44)
[E]     at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:72)
[E]     at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:52)
[E]     at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:136)
[E]     at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:34)
[E]     at cats.effect.internals.IOBracket$.$anonfun$apply$1(IOBracket.scala:44)
[E]     at cats.effect.internals.IOBracket$.$anonfun$apply$1$adapted(IOBracket.scala:34)
[E]     at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:341)
[E]     at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:119)
[E]     at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:355)
[E]     at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:376)
[E]     at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:316)
[E]     at cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
[E]     at cats.effect.internals.PoolUtils$$anon$2$$anon$3.run(PoolUtils.scala:51)
[E]     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[E]     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[E]     at java.base/java.lang.Thread.run(Thread.java:834)
[E] Caused by: scala.UninitializedFieldError: Uninitialized field: /home/cbrown/projects/franklin/application/src/main/scala/com/azavea/franklin/api/endpoints/SearchEndpoints.scala: 20
[E]     at com.azavea.franklin.api.endpoints.SearchEndpoints$.searchFiltersValidator(SearchEndpoints.scala:20)
[E]     at com.azavea.franklin.api.endpoints.SearchEndpoints$.<init>(SearchEndpoints.scala:21)
[E]     at com.azavea.franklin.api.endpoints.SearchEndpoints$.<clinit>(SearchEndpoints.scala)
[E]     ... 29 more
```

So going to leave it as-is for now

File: application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala (hunk 2 above)
Comment (notthatbreezy) [reply application/src/main/scala/com/azavea/franklin/crawler/StacImport.scala]: Couldn't get this to work, opened https://github.com/azavea/azavea.g8/issues/4 to figure out how to get it to work

**Pull request merged**